src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -376,32 +403,43 @@ def read_file(
                 if os.path.isfile(os.path.join(stage_source_dir_path, f))
             ]
 
-        # TODO: SNOW-1253672, there is a bug in the non-local testing code that
-        #  snowflake.snowpark.dataframe_reader.DataFrameReader._infer_schema_for_file_format does not
-        #  take PATTERN into account, when inferring schema from multiple files
-        # pattern = options.get("PATTERN") if options else None
-        #
-        # if pattern:
-        #     local_files = [
-        #         f
-        #         for f in local_files
-        #         if re.match(pattern, f[: -len(StageEntity.FILE_SUFFIX)])
-        #     ]
-
         file_format = format.lower()
         if file_format in SUPPORT_READ_OPTIONS:
             for option in options:
-                if option not in SUPPORT_READ_OPTIONS[file_format]:
+                if str(options[option]).upper() == "NONE":
```

Comment:
NIT: Let's define something like `supported_options_for_format = SUPPORT_READ_OPTIONS[file_format]` to simplify this bit more.